contrib/cmake/build3rdParty/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+
+
+cmake_minimum_required(VERSION 3.13)
+project(Build3rdParty)
+
+if (WIN32)
+    # accurate __cplusplus macro for vc++, selecting c++17 here for windows builds though the MEGA SDK library must build for older c++ standards also
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:__cplusplus /std:c++17")
+endif()
+
+if (APPLE)
+    set(CMAKE_CXX_STANDARD 11)
+    SET(CMAKE_CXX_STANDARD_REQUIRED ON)
+endif()
+
+add_executable(build3rdParty build3rdParty.cpp)
+if (NOT WIN32 AND NOT APPLE)
+    target_link_libraries(build3rdParty stdc++fs)
```

Comment:
Do we support building with libc++? This becomes a bit more complicated if so

Reply:
We should support them all, I think, but probably we just haven't run into any issue with this one yet.   Actually we can't really use std::filesystem for production stuff since we are on C++11 for now.   However it is very convenient and we use it in some test apps etc, referencing the early not-quite-standard experimental versions that were available with the C++11 compilers.